tt_eager/tt_dnn/op_library/bmm/multi_core_reuse_mcast_2d_optimized/bmm_op_multi_core_reuse_mcast_2d_optimized.cpp

```diff
@@ -176,27 +137,20 @@ operation::ProgramWithCallbacks create_program_mcast_in0_in1(
     uint32_t in0_sender_num_cores_along_width;
 
     // Only used for in0 block sharded
-    CoreRange in0_mcast_cores_with_work_and_in_receiver_grid(
-        {(std::size_t)start_core_x, (std::size_t)start_core_y},
-        {(std::size_t)start_core_x, (std::size_t)start_core_y});  // Placeholder
-    CoreRange in0_mcast_cores_without_work_and_not_in_receiver_grid(
-        {(std::size_t)start_core_x, (std::size_t)start_core_y},
-        {(std::size_t)start_core_x, (std::size_t)start_core_y});  // Placeholder
+    std::optional<CoreRange> in0_mcast_cores_without_work_and_not_in_receiver_grid;
     if (in0_block_sharded) {
-        in0_mcast_cores_with_work_and_in_receiver_grid = all_cores_with_work;
-
         CoreCoord in0_shard_grid = in0_buffer->shard_spec().grid().bounding_box().grid_size();
         // in0 shard grid already accounts for transpose_mcast
         // ie. If transpose_mcast, in0 width is along y
         in0_sender_num_cores_along_width = transpose_mcast ? in0_shard_grid.y : in0_shard_grid.x;
         if (in0_sender_num_cores_along_width > num_blocks_x) {
             in0_mcast_cores_without_work_and_not_in_receiver_grid =
                 transpose_mcast ? CoreRange(
-                                      {(std::size_t)start_core_x, (std::size_t)start_core_y + num_blocks_x - 1},
+                                      {(std::size_t)start_core_x, (std::size_t)start_core_y + num_blocks_x},
```

Comment:
what is the behaviour before and the behaviour after for changing the core range from having  `start_core_y + num_blocks_x - 1` vs `start_core_y + num_blocks_x`?

Reply:
When you create a kernel on a core that already has a kernel loaded:
- if the RISC, NOC, kernels are all the same, then it will silently do nothing (my evidence is, I printed the compile time args in the loaded kernel and it had the values of the compile time args the first time it was loaded)
- if kernels specs are not the same, I think it will error out

In the first case, this should also error out

Reply:
Ok. Thanks.